lib/maplibre-compose-material3/src/androidMain/kotlin/dev/sargunv/maplibrecompose/material3/util/NumberFormatter.android.kt

```diff
@@ -0,0 +1,14 @@
+package dev.sargunv.maplibrecompose.material3.util
+
+import androidx.compose.ui.text.intl.Locale
+import java.text.NumberFormat
+
+internal actual class NumberFormatter
+actual constructor(locale: Locale, maximumFractionDigits: Int) {
+  private val format =
+    NumberFormat.getInstance(locale.platformLocale).also {
+      it.maximumFractionDigits = maximumFractionDigits
+    }
+
+  actual fun format(value: Number): String = format.format(value)
+}
```

Comment:
I did use the previous method because the docs say the expect-actual classes are in beta and because it mentions

> You might have already seen this approach in demonstration materials. However, using classes in simple cases where interfaces would be sufficient is not recommended.

https://www.jetbrains.com/help/kotlin-multiplatform-dev/multiplatform-expect-actual.html#expected-and-actual-classes

(But to be honest, I like your approach more, despite the "not recommended" text in the docs)